File: javascript/Intermediate-Algorithm-Scripting/convertHTMLEntities.js
/*

Convert the characters &, <, >, " (double quote), and ' (apostrophe), in a string to their corresponding HTML entities.

*/

function convertHTML(str) {
    const replace = {
      '>': '&​gt;',
      '<': '&​lt;',
      "'":  '&​apos;',
      '"': '&​quot;',
      '&': '&​amp;',
    };
    const keys = Object.keys(replace);
  
    return str.split('').map(char=>{
        return keys.indexOf(char) != -1 ? replace[char] : char;
    })
    .join("");

}
  
convertHTML("Dolce & Gabbana");

File: python/Intermediate-Algorithm-Scripting/convertHTMLEntities.py
'''

Convert the characters &, <, >, " (double quote), and ' (apostrophe), in a string to their corresponding HTML entities.

'''

def convertHTML(string):
    replace = {
      '>': '&gt;',
      '<': '&lt;',
      "'":  '&apos;',
      '"': '&quot;',
      '&': '&amp;',
    }
    return ''.join(replace.get(c, c) for c in string)
  



  
convertHTML("Dolce & Gabbana")